Dockerfile

```diff
@@ -16,6 +16,18 @@ ENV MAXIMAPOOL=/opt/maximapool \
 ENV MAXIMA_LOCAL_PATH=${BUILD_FOR_MOODLE:+moodle-qtype_stack/stack/maxima}
 ENV MAXIMA_LOCAL_PATH=${MAXIMA_LOCAL_PATH:-assStackQuestion/classes/stack/maxima}
 
+RUN apt-get update \
+    && apt-get install -y \
+      openjdk-$(echo "$JAVA_VERSION" | cut -d '-' -f2 | cut -d '.' -f1)-jdk \
```

Comment:
This is different from the old way (see below). I have no idea how the old way could have ever worked. The `JAVA_VERSION` environment variable looks like `jdk-11.0.18+10` and the `%%` thing extracts text from the start of the string up to the first instance of "not a string of digits in the range 0-9". For example:

```
$ BLAH="1234asdf"
$ echo  echo "${BLAH%%[!0-9]*}"
1234
```

What we want to extract from the `jdk-11.0.18+10`  string is the `11` part and there's just no way the old code can do that because it's specifically asking for things that are not numbers. Maybe there's a better way than what I've done here, but at least it works.